src/test/java/com/bobocode/hoverla/bring/context/BeanConfigClassValidatorTest.java

```diff
@@ -0,0 +1,101 @@
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+class BeanConfigClassValidatorTest {
+
+    private BeanConfigClassValidator validator;
+
+    @SneakyThrows
+    private static Stream<Arguments> invalidBeanConfigs() {
+        return Stream.of(
+                Arguments.of(
+                        AbstractTestBeanConfig.class,
+                        "Configuration class must not be an abstract class",
+                        "Doesn't allow bean config class to be abstract"),
+                Arguments.of(EnumTestBeanConfig.class,
+                        "Configuration class must not be an enum",
+                        "Doesn't allow bean config class to be an enum"),
+                Arguments.of(InterfaceTestBeanConfig.class,
+                        "Configuration class must not be an interface",
+                        "Doesn't allow bean config class to be an interface"),
+                Arguments.of(NoDefaultConstructorTestConfig.class,
+                        "Configuration class must have one public no-arguments constructor",
+                        "Doesn't allow bean config class with no public no-arg constructor"),
+                Arguments.of(Class.forName("com.bobocode.hoverla.bring.testsubject.validator.config.NonPublicTestConfig"),
+                        "Configuration class must be public",
+                        "Doesn't allow non-public bean config class"),
+                Arguments.of(RecordTestBeanConfig.class,
+                        "Configuration class must not be a record",
+                        "Doesn't allow bean config class to be a record"),
+                Arguments.of(InnerClassTestBeanConfig.InnerTestBeanConfig.class,
+                        "Configuration class must not have an enclosing class",
+                        "Doesn't allow bean config class to be inner or nested")
+        );
+    }
+
+    @SneakyThrows
+    public static Stream<Arguments> invalidBeanMethodsMessages() {
+        return Stream.of(
+                Arguments.of("privateMethod method must be public",
+                        "Doesn't allow method marked as @Bean to be private"),
+                Arguments.of(
+                        "packagePrivateMethod method must be public",
+                        "Doesn't allow method marked as @Bean to be package-private"),
+                Arguments.of(
+                        "protectedMethod method must be public",
+                        "Doesn't allow method marked as @Bean to be protected"),
+                Arguments.of(
+                        "staticMethod method must not be static",
+                        "Doesn't allow method marked as @Bean to be static")
+        );
+    }
+
+    @BeforeEach
```

Comment:
I would recommend to create only 1 test instance, since you're using `validator` in every test. Thus let's annotate this class with `@TestInstance(TestInstance.Lifecycle.PER_CLASS)` and either remove this `setUp()` method and make a direct assignment on line `27` or leave it but change to `@BeforeAll`. And also don't forget to remove all `static` modifiers.

Reply:
IMO, this is something like preliminary optimization, which is not necessary.

I mean, in case we add to validator some inner state like cache one day, then we'll have to change this part. I've never seen people would do such things, I guess it's really more about test run speed optimization

Can you tell me any real advantage of it?

PS: I'll remove static methods for consistency, but it's more about style anyway

Reply:
I agree with Nazar. It's validator and state of it's class never changes.

Reply:
And that made me think like, do we really need it for ApplicationContextImpl? 
Yes, we use the same context, but we don't get a strong guarantee that some of the tests couldn't change the state and don't screw everything up.

Reply:
Okay, I'll add it for now, don't want to argue very much, for now it maybe makes sense

Reply:
PS: but it seems like it makes sense to add it to every test in this PR, doesn't it?

Reply:
Fixed

Reply:
@DanielChanger well this feature may only be used when we`re not dependent on state, just like in this case. And it is also an optimization (only 1 instance gets created). In case we add smth like cache some day, this test will definitely be rewritten. But I don't see such possibility in the nearest future)

Reply:
@DanielChanger I also think about ApplicationContextImpl and in this case you're right. Because context container is there and some test can change it. So I want to fix test class